design-proposals/klip-41-ksqldb-.net-client.md

```diff
@@ -0,0 +1,72 @@
+# KLIP 41 - ksqlDB .NET Client
+
+**Author**: Alex Basiuk (@alex-basiuk) | 
+**Release Target**: TBD | 
+**Status**: _In Discussion_ | 
+**Discussion**: TBD
+
+**tl;dr:** _Create a ksqlDB client for .NET which is on par with the existing Java client.
+The .NET community will benefit from the this project. The goal is to provide first class developer experience and 
+simplify writing event streaming applications in .NET._
+
+## Motivation and background
+
+Basically, it's similar to [KLIP 15](./klip-15-new-api-and-client.md).
+It should lower the barrier for .NET developers and increase adoption of ksqlDB by the community.
+The library should offer the same feature set as the Java library and should follow common .NET conventions and patterns.  
+
+## What is in scope
+
+* A new client implemented in C#.
+* Initially, the client will be targeted at .NET Core 3 and .NET Framework 5. Support of older versions will be added later.
+* The client will support execution and streaming of queries (both push and pull), inserting of rows into streams, DDL operations and admin operations
+such as list and describe.
+* The client will utilise C# [async streams](https://docs.microsoft.com/en-us/dotnet/csharp/tutorials/generate-consume-asynchronous-stream) feature.
+* The client will also support a direct / synchronous interaction style.
+* High-quality documentation and examples.
```

Comment:
Super. :) If this is going to live inside of a repo owned by you for a while, we can put the documentation in a README there so it is self-contained.